specs/CustomScrollbar.md

```diff
@@ -0,0 +1,93 @@
+Custom Scrollbar Style
+===
+
+# Background
+Developers would like to be able to set the scrollbar to Windows fluent overlay style and 
+match the Windows theme. Some customers have used browser flags but they change names and behavior and are not reliable. Thus we want
+to provide an extendable API that allows developers to set scrollbar style. Currently, 
```

Comment:
@david-risney It occurred to me that the extensibility does have a problem: Suppose we later add a new scrollbar style, call it `Awesome`. How does the C++ client know whether or not the current CoreWebView2 supports Awesome? The C++ client builds the `CoreWebView2EnvironmentOptions` in-proc from its own internal library, so its call to `put_ScrollBarStyle(Awesome)` succeeds. But it doesn't find out until `CoreWebView2Environment.CreateAsync(options)` fails with `E_INVALIDARG`. 

At this point, the app isn't sure which of the 25 options is invalid. It could try scaling back each option until the call succeeds, but you don't know which order to scale them back. 

Most of the CoreWebView2EnvironmentOptions are just ignored if they aren't supported (because the runtime never queries for them). This one is weird because the option is an enum, which means that the runtime may see a value it doesn't like.

I guess what you have to do is start with a minimal version, and then slowly add more stuff. "Okay, first I'll try ColorScheme=Light. Cool that worked. How about ColorScheme=MorningFog? Nope, didn't work. Okay I'll stick with Light. Next to see what I can configure for touch targeting. First, I'll try TouchTargeting=Simple. Okay that worked. How about TouchTargeting=Tether? Nope, didn't work. Okay, so I'll stick with Simple. Now to scroll bars..."

You could end up creating a ton of temporary Environment objects while trying to probe which features you can actually use.

Reply:
What if we don't fail WV2 creation if an non-exist scrollbar was used? Just fall back to Default?

Reply:
This is an existing issue we have the EnvironmentOptions but previously we've only been looking at it for EnviromentOptions.AdditionalBrowserArguments (real feedback from partners). I'm going to talk with @aluhrs13 about this to see how we might accommodate additional properties on the EnvironmentOptions. Thanks.

Reply:
Agreed to track this separately as we have similar issue in the past.